internal/semver/semver.go

```diff
@@ -39,28 +50,41 @@ type version struct {
 	// number. If there is no numeric part in the pre-release segment, this
 	// field is nil.
 	PrereleaseNumber *int
+	// SpecVersion is the SemVer spec version detected in the version string.
+	// Currently, this only impacts stringifying of prerelease numbers.
+	SpecVersion string
 }
 
-// semverV1PrereleaseNumberRegexp extracts the prerelease number, if present, in
-// the prerelease portion of the SemVer 1.0.0 version string. For example, a
-// version string like "1.2.3-alpha01" is a SemVer 1.0.0. compliant, numbered
-// prerelease - https://semver.org/spec/v1.0.0.html#spec-item-4.
-var semverV1PrereleaseNumberRegexp = regexp.MustCompile(`^(.*?)(\d+)$`)
+var (
+	// semverV1PrereleaseNumberRegexp extracts the prerelease number, if present, in
+	// the prerelease portion of the SemVer 1.0.0 version string. For example, a
+	// version string like "1.2.3-alpha01" is a SemVer 1.0.0. compliant, numbered
+	// prerelease - https://semver.org/spec/v1.0.0.html#spec-item-4.
+	semverV1PrereleaseNumberRegexp = regexp.MustCompile(`^(.*?)(\d+)$`)
+
+	// ErrInvalidVersion is returned when the version string provided is invalid as
+	// per the SemVer spec - https://semver.org.
+	ErrInvalidVersion = errors.New("invalid version format")
```

Comment:
I don't think this is used outside of this package, in which case I would unexport it